cpp/src/rolling/jit/kernel.cu

```diff
@@ -51,8 +51,8 @@ __global__ void gpu_rolling_new(cudf::size_type nrows,
                                 FollowingWindowType following_window_begin,
                                 cudf::size_type min_periods)
 {
-  cudf::thread_index_type i            = blockIdx.x * blockDim.x + threadIdx.x;
-  cudf::thread_index_type const stride = blockDim.x * gridDim.x;
+  auto i            = cudf::detail::grid_1d::global_thread_id();
+  auto const stride = cudf::detail::grid_1d::grid_stride();
```

Comment:
Reverting changes in `rolling/jit/kernel.cu` since it would need another `#include` that I suspect might not work in JIT code.
```suggestion
  cudf::thread_index_type i = blockIdx.x * blockDim.x + threadIdx.x;
  cudf::thread_index_type const stride = blockDim.x * gridDim.x;
```